bluemira/equilibria/coils/_field.py

```diff
@@ -1026,14 +822,18 @@ def _combined_control(
         points = inside[..., 0]
 
         if np.any(~points):
-            response[~points] = greens_func(x[~points], z[~points])
+            response[~points] = self._response_greens(
+                greens_func, x[~points], z[~points]
+            )
 
         if np.any(points):
-            response[points] = semianalytic_func(x[points], z[points])
+            response[points] = self._response_analytical(
+                semianalytic_func, x[points], z[points]
+            )
 
         return response
 
-    def _B_response_analytical(
+    def _response_analytical(
```

Comment:
I think I may have missed something, but can we not use _response_analytical from CoilGroupFieldsMixin in CoilFieldsMixin's _combined_control?

Reply:
Those three functions on CoilFieldsMixin are needed to get the right shaped array into other functions if you are using just the one coil (not wrapped in a coilgroup). 

It might be possible to adjust this but eg the x z positions are not arrays on Coils whereas they are on all of the grouping objects. This was so that you could do coil.x and get a number and not an array which at the time was a desired thing. 

I did try previously to reduce this but this was the best I could do.